docs/DeveloperGuide.md

```diff
@@ -2,33 +2,82 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Class diagram for v2.0
+![Class Diagram](assets/classdiagram.svg)
```

Comment:
maybe you can leave out less important details and make the class diagram a bit bigger so it is more readable?